### ISSUE TITLE
Any example for an optimized architecture search for an entire CNN Model?

### ISSUE THREAD

**Issue by gyb1325** [NVlabs/timeloop#14]
Thanks for providing such a tool for us. That is very helpful. 

I am wondering could you provide an example for searching an optimized acc architecture for an entire model. It seems that current examples only for a specific CNN layer. 

Thank you so much~

**Comment by angshuman-parashar** [comment]
Thank you for your interest. To evaluate an entire model, you would invoke Timeloop serially, once for each layer. Take a look at the `scripts/` directory, where we have provided a set of Python wrappers to invoke Timeloop and to parse the results. Take a look at the `sample.py` and `cnn_layers.py` files in that directory. Of course, because you are evaluating each layer independently, Timeloop will not perform cross-layer optimizations (fusion). Support for fusion is planned for a future update.

**Comment by cnjsdfcy** [comment]
Hi Angshuman,

As for the cross-layer optimizations, I think it is not limited only to fusion at algorithm level, like fuse two algorithm operations, but also the reuse of memory components, like the SRAM between outer DDR and inner registers, just like the GlobalBuffer in  architecture of example 3 in Timeloop tuturial. Effiently use of this kind inter-mediate buffer is critial to overall inference perfomance.
So basically, this is a scheduling mechanism of data movement into buffer or out from it, I believe this problem is also belonged to mapping space searching.
Are you planning to support this kind mapping search for entire networks?

Thanks~

**Comment by angshuman-parashar** [comment]
You said "not limited only to fusion at algorithm level, like fuse two algorithm operations". If you are talking about one operation as in a network layer (for example), then Timeloop already models data reuse across multiple levels of the memory hierarchy, and the mapper explores all possible ways to tile data across the hierarchy. The word "fusion" applies only when you are considering producer-consumer relationship across multiple operators. And yes, the interplay of fusion with multi-level tiling is indeed interesting. Modeling fusion and exploring the mapspace of fused operations is high on our priority list of future updates for Timeloop. Please stay tuned.

**Comment by cnjsdfcy** [comment]
Thanks for your reply, looking forward to this updates.
And again, thanks for creating and sharing Timeloop tool.
